rkt/list.go

```diff
@@ -61,40 +68,65 @@ func runList(cmd *cobra.Command, args []string) (exit int) {
 			// TODO(vc): we should really hold a shared lock here to prevent gc of the pod
 			pmf, err := p.readFile(common.PodManifestPath(""))
 			if err != nil {
-				stderr("Unable to read pod manifest: %v", err)
+				errors = append(errors, newPodListReadError(p, err))
 				return
 			}
 
 			if err := pm.UnmarshalJSON(pmf); err != nil {
-				stderr("Unable to load manifest: %v", err)
+				errors = append(errors, newPodListLoadError(p, err, pmf))
 				return
 			}
 
 			if len(pm.Apps) == 0 {
-				stderr("Pod contains zero apps")
+				errors = append(errors, newPodListZeroAppsError(p))
 				return
 			}
 		}
 
-		uuid := ""
-		if flagFullOutput {
-			uuid = p.uuid.String()
-		} else {
-			uuid = p.uuid.String()[:8]
+		type printedApp struct {
+			uuid    string
+			appName string
+			imgName string
+			state   string
+			nets    string
 		}
 
-		for i, app := range pm.Apps {
+		appsToPrint := []printedApp{}
+		uuid := p.uuid.String()
+		state := p.getState()
+		nets := fmtNets(p.nets)
+		if !flagFullOutput {
+			uuid = uuid[:8]
+		}
+		for _, app := range pm.Apps {
 			// Retrieve the image from the store.
-			im, err := s.GetImageManifest(app.Image.ID.String())
+			imj, err := s.GetImageManifestJSON(app.Image.ID.String())
 			if err != nil {
-				stderr("Unable to load image manifest: %v", err)
+				errors = append(errors, newPodListImageStoreFailure(p, err, &pm, app))
+				return
 			}
-
-			if i == 0 {
-				fmt.Fprintf(tabOut, "%s\t%s\t%s\t%s\t%s\n", uuid, app.Name.String(), im.Name.String(), p.getState(), fmtNets(p.nets))
-			} else {
-				fmt.Fprintf(tabOut, "\t%s\t%s\t\t\n", app.Name.String(), im.Name.String())
+			var im *schema.ImageManifest
+			if err = json.Unmarshal(imj, &im); err != nil {
+				errors = append(errors, newPodListImageLoadFailure(p, err, &pm, imj, app))
+				return
 			}
+
+			appsToPrint = append(appsToPrint, printedApp{
+				uuid:    uuid,
+				appName: app.Name.String(),
+				imgName: im.Name.String(),
+				state:   state,
+				nets:    nets,
+			})
+			uuid = ""
+			state = ""
+			nets = ""
```

Comment:
Can you make `uuid`, `state`, `nets` variables local to the `for` loop so that you don't have to clear them?